django/db/migrations/operations/fields.py

```diff
@@ -49,12 +49,17 @@ def references_field(self, model_name, name, app_label):
         if model_name_lower == self.model_name_lower:
             if name == self.name:
                 return True
-            elif (
-                self.field
-                and hasattr(self.field, "from_fields")
-                and name in self.field.from_fields
-            ):
-                return True
+            if self.field:
+                if (
+                    hasattr(self.field, "from_fields")
+                    and name in self.field.from_fields
+                ):
+                    return True
+                elif (
+                    self.field.generated
+                    and name in Q(self.field.expression).referenced_base_fields
```

Comment:
Ditto, it seems wrong to create a `Q` object for the sole purpose of calling `referenced_base_fields` so `get_children_from_q` can immediately call `get_paths_from_expression`.